frontend/ui/widget/imageviewer.lua

```diff
@@ -417,6 +417,16 @@ function ImageViewer:_new_image_wg()
         rotation_angle = rotate_clockwise and 90 or 270
     end
 
+    -- Rotate if image and screen aspect ratio are misaligned
+    local image_fit_rotate = tonumber(G_reader_settings:readSetting("image_fit_rotate", 0))
+    local aspect_screen = Screen:getWidth() / Screen:getHeight()
+    local aspect_image = self.image:getWidth() / self.image:getHeight()
+    if (aspect_screen <= 1 and aspect_image > 1) or
+        (aspect_screen > 1 and aspect_image <= 1) then
+        rotation_angle = (rotation_angle - image_fit_rotate) % 360
+    end
```

Comment:
In the section just above (line 407), we have the current rotation status remembered as `self.rotated`, and we use this to pick up a totation_angle. You then override or tweak the rotation_angle this has picked.
I think it would be clearer/saner from a UX perspective if you just, on launch, if your aspect ratio checks decide it should be rotated, to just set `self.rotated=true` so the bottom button about Rotation says the truth :)